src/sentry/api/endpoints/organization_vitals_overview.py

```diff
@@ -29,39 +51,70 @@ def get(self, request: Request, organization: Organization) -> Response:
         # try to get all the projects for the org even though it's possible they don't have access
-        project_ids = Project.objects.filter(
-            organization=organization, status=ProjectStatus.VISIBLE
-        ).values_list("id", flat=True)[
-            0:1000
+        projects = Project.objects.filter(organization=organization, status=ProjectStatus.VISIBLE)[
+            0:MAX_PROJECTS
         ]  # only get 1000 because let's be reasonable
 
-        # TODO: add logic to make sure we only show for orgs with 100+ relevant transactions
-        # for each category
+        project_ids = list(map(lambda x: x.id, projects))
 
-        # Web vitals: p75 for LCP and FCP
-        # Mobile vitals: Cold Start and Warm Start
-        with self.handle_query_errors():
+        def get_discover_result(columns, referrer):
             result = discover.query(
                 query="transaction.duration:<15m transaction.op:pageload event.type:transaction",
-                selected_columns=[
-                    "p75(measurements.lcp)",
-                    "p75(measurements.fcp)",
-                    "measurements.app_start_cold",
-                    "measurements.app_start_warm",
-                ],
-                limit=1,
+                selected_columns=columns,
+                limit=MAX_PROJECTS,
                 params={
                     "start": timezone.now() - timedelta(days=7),
                     "end": timezone.now(),
                     "organization_id": organization.id,
                     "project_id": list(project_ids),
                 },
-                referrer="api.organization-vitals",
+                referrer=referrer,
             )
-            # only a single result
-            data = result["data"][0]
-            # map the names
+            return result["data"]
+
+        with self.handle_query_errors():
+            org_data = get_discover_result(BASIC_COLUMNS, "api.organization-vitals")
+            # no data at all for any vital
+            if not org_data:
+                return self.respond(
+                    {
+                        "FCP": None,
+                        "LCP": None,
+                        "appStartWarm": None,
+                        "appStartCold": None,
+                        "fcpCount": 0,
+                        "lcpCount": 0,
+                        "appColdStartCount": 0,
+                        "appWarmStartCount": 0,
+                        "projectData": [],
+                    }
+                )
             output = {}
-            for key, val in data.items():
+            # only a single result
+            for key, val in org_data[0].items():
                 output[NAME_MAPPING[key]] = val
+
+            # get counts by project
+            project_data = get_discover_result(
+                ["project_id"] + BASIC_COLUMNS, "api.organization-vitals-per-project"
+            )
+
+            # check access for project level data
```

Comment:
Why does this check happen after the `per-project` query? Why not filter the project IDs down before querying Clickhouse?

Reply:
@evanh I specifically do this so we can cache on a per-organization level regardless of team membership. Even though the query will be more expensive, my thought is that overall it will be cheaper if we get more cache hits (and the impact of caching will be larger for larger organizations).

Reply:
I think in practical terms that assumption will depend on how many projects you are filtering out and how long you are caching. I would suggest gathering some metrics on that, because if you are querying for 1000 projects and regularly filtering down to 1 (or 0? is that possible?) you are putting a lot of unnecessary pressure on Clickhouse.

Also, are the queries for the project IDs above sorted implicitly? That might break your caching.

Reply:
@evanh I am planning on caching for 24 hours. I can also reduce the maximum number of projects if you think that helps, we are mostly targeting low-mid sized companies so that's totally an option.

> Also, are the queries for the project IDs above sorted implicitly? That might break your caching.
The projects are sorted implicitly since I am going to use the organization ID alone as the cache key, then it wouldn't matter right? I can also explicitly sort if that helps but I don't think it would.

Reply:
@evanh Ok, I lowered the limit of the number of projects from 1000 to 100. If an organization has 100+ projects, we will just give up and not even query snuba. Hopefully this will reduce the load on ClickHouse while not excluding too many folks.

Reply:
Lowered it to 25 after talking to AJ

Reply:
You don't have to exclude orgs with >N projects, Clickhouse can handle that load. I was just pointing out a potential performance problem.

If the projects are not sorted, then isn't it possible every time this endpoint is hit, a different set of N projects will be fetched? In that case the results would not be deterministic. That's what I was driving at. Caching by org_id makes total sense to me.